core/invoker/src/main/scala/whisk/core/invoker/Invoker.scala

```diff
@@ -87,6 +89,38 @@ object Invoker {
 
     val msgProvider = SpiLoader.get[MessagingProvider]
     val producer = msgProvider.getProducer(config, ec)
+
+    val invokerUUID = sys.env.getOrElse("INVOKER_ASSIGNED_UUID", UUID().toString)
+    val proposedInvokerId = if (args.length == 1) { args(0).toInt } else { -1 }
```

Comment:
True with current simplistic code.  I could document/enforce that if we think it is important to do so.

I think where we this is likely to end up is that the Registrar actor would get a little more complex and interact with the rest of the load balancing subsystem.  In particular, instead of blindly incrementing a counter if there is not match and no proposed id, it could first look to see if there is an abandoned topic (because of an unhealthy/offline invoker) and then prefer to assign that topic to the invoker instead of creating a new topic.

Reply:
I added a check to the code to insist that either INVOKER_ASSIGNED_UUID is defined or a proposedId is provided on the command line.  If neither is provided, we abort the invoker.